reverse_only_letters.py

```diff
@@ -0,0 +1,44 @@
+#Problem: https://leetcode.com/problems/reverse-only-letters/
+
+# Time complexity: O(N)
+# Memory complexity: O(N)
+#Method 1:
+def reverse_letters(s):
+    chars = []
+    reverse_letters = []
+    for ch in s:
+        if ch.isalpha():
+            chars.append(ch)
+
+    for ch in s:
+        if ch.isalpha():
+            reverse_letters.append(chars.pop())
+        else:
+            reverse_letters.append(ch)
+
+    return ''.join(reverse_letters)
+            
+
+# Time complexity: O(N)
+# Memory complexity: O(N)
+#Method 2:
+def rev_letters(s):
+    i = 0
+    j = len(s) - 1
+    letters = list(s)
```

Comment:
That works but it requires you to do ''.join(letters), which is not ideal.

Try using a type of string here (letters = "").

Tip: You can pre-populate your string to easily manipulate it.